src/Simulation/Simulators.Tests/ResourcesEstimatorTests.cs

```diff
@@ -1,196 +1,236 @@
+        public void VerifyDataTest()
+        {
+            var sim = new ResourcesEstimator();
+
+            VerySimpleEstimate.Run(sim).Wait();
+            var data = sim.Data;
+
+            Assert.Equal(1.0, data.Rows.Find("CNOT")["Sum"]);
+            Assert.Equal(0.0, data.Rows.Find("R")["Sum"]);
+            Assert.Equal(2.0, data.Rows.Find("QubitClifford")["Sum"]);
+            Assert.Equal(3.0, data.Rows.Find("Width")["Sum"]);
+        }
+
+        /// <summary>
+        /// Verifies the calls ToCSV returns a non-empty string.
+        /// </summary>
+        [Fact]
+        public void ToTSVTest()
+        {
+            var sim = new ResourcesEstimator();
+
+            VerySimpleEstimate.Run(sim).Wait();
+            var data = sim.ToTSV();
+            Console.WriteLine(data);
+
+            Assert.NotNull(data);
+            var rows = data.Split('\n');
+            Assert.Equal(10, rows.Length);
+
+            var cols = rows[0].Split('\t');
+            Assert.Equal("Metric", cols[0].Trim());
+            Assert.Equal(3, cols.Length);
+
+            var cliffords = rows.First(r => r.StartsWith("QubitClifford")).Split('\t');
+            Assert.Equal(3, cliffords.Length);
+            Assert.Equal("2", cliffords[1]);
         }
 
-        /// <summary>
-        /// Shows that T gates on different qubits are counted for depth purposes as 
-        /// executing in parallel.
-        /// </summary>
-        [Fact]
-        public void DepthDifferentQubitsTest()
-        {
-            var sim = new ResourcesEstimator();
-
-            // using(q = Qubit[3]) { T(q[0]); T(q[1]); T(q[3]); T(q[0]); }
-            DepthDifferentQubits.Run(sim).Wait();
-            var data = sim.Data;
-
-            Assert.Equal(4.0, data.Rows.Find("T")["Sum"]);
-            Assert.Equal(3.0, data.Rows.Find("Width")["Sum"]);
-            Assert.Equal(2.0, data.Rows.Find("Depth")["Sum"]);
+        /// <summary>
+        /// Shows that T gates on different qubits are counted for depth purposes as 
+        /// executing in parallel.
+        /// </summary>
+        [Fact]
+        public void DepthDifferentQubitsTest()
+        {
+            var sim = new ResourcesEstimator();
+
+            // using(q = Qubit[3]) { T(q[0]); T(q[1]); T(q[3]); T(q[0]); }
+            DepthDifferentQubits.Run(sim).Wait();
+            var data = sim.Data;
+
+            Assert.Equal(4.0, data.Rows.Find("T")["Sum"]);
+            Assert.Equal(3.0, data.Rows.Find("Width")["Sum"]);
+            Assert.Equal(2.0, data.Rows.Find("Depth")["Sum"]);
+        }
+
+        [Fact]
+        public void QubitReuseWithOptimizedDepthTest() {
+            QCTraceSimulators.QCTraceSimulatorConfiguration config = ResourcesEstimator.RecommendedConfig();
+            config.OptimizeDepth = true;
+            var sim = new ResourcesEstimator(config);
+
+            QubitReuseWithOptimizedDepth.Run(sim).Wait();
+
+            Assert.Equal(2.0, sim.Data.Rows.Find("QubitCount")["Sum"]);

```

Comment:
What is the "QubitCount"? From the whitepaper my understanding was that for OptimizeDepth = true config we should produce two numbers: optimal minimal depth and compatible, as low as we can get, width.

Reply:
QubitCount is the old value we used to have and I didn't remove it to keep compatibility. It's also more reasonable per-scope than width. Width is useful mostly for the entire program.